### DISCUSSION TITLE
Allow running scripts locally (without unnecessary external content)

### DISCUSSION THREAD

**Discussion by tcarreira** [community-scripts/ProxmoxVE#11] [Ideas]
It would be straight forward to `git clone` this repo and the run intended script directly. 

Unfortunately, most script are currently sourcing content from the internet, like
```
source <(curl -s https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/misc/build.func)
```

I think it would be nice if I could just run/modify my local files and trust it is using the local files for everything.

( I can contribute, but I would like to know if the community feels the same first)

**Comment by Mellowlynx** [comment]
You can do this by cloning it to a local repository.
The external files have the basic functions that are used by most/all scripts.
This way, it's easier to make new scripts or update basic functions just once.

Edit:
This also makes it easier to check the scripts for malicious code.
When build.func (or one of the two other reused scripts) gets an update, that the time to be on high alert.

**Reply by LightTemplar** [reply]
> malicious code. 
> the time to be on high alert.

Were there already such cases?

**Comment by tcarreira** [comment]
Thanks for the input/feedback. I will close the discussion then
